app/helpers/upload_helper.rb

```diff
@@ -1,14 +1,19 @@
 # Helpers for handling uploading data files for various models.
 module UploadHelper
-  def process_file_upload
+  def process_file_upload(allowed_filetypes = %w[.csv .yaml .yml])
     encoding = params[:encoding] || 'UTF-8'
     upload_file = params.require(:upload_file)
 
     if upload_file.size == 0
       raise StandardError, I18n.t('upload_errors.blank')
     end
 
-    filetype = File.extname(upload_file.original_filename)
+    if allowed_filetypes.size == 1
```

Comment:
This check is a little awkward because there are two possible extensions for YAML. This code will work okay if you always expect users to pass `'.yml'` (but not `'.yaml'`). You can communicate this by adding to this method docstring and by removing `'.yaml'` from the default argument value